staging/vhost-device-gpu/src/device.rs

```diff
@@ -110,10 +113,14 @@ pub struct VhostUserGpuBackend {
     shm_region: Option<VirtioShmRegion>,
 }
 
+pub struct VhostUserGpuBackend {
+    inner: Mutex<VhostUserGpuBackendInner>,
```

Comment:
Yes the mutex increases performance overhead, but we were doing the same thing before (we had an RwLock, but same principle applies, it was just around the whole struct not inside it). This is unfortunate but the `vhost-user-backend` crate requires the trait to be Sync, even if we have 1 worker thread. I feel like the API choices of that crate complicates the code for `vhost-user-gpu` and potentially introduces performance overhead for us 🙁.

I decided to use `Mutex` instead of `RwLock`, because now the methods that only read don't really need a lock at all (e.g.  `num_queues()`) and methods that may write just use the mutex. 

The whole point of doing this was to avoid the deadlock that I ran into when I tried to call `VringEpollHandler::register_listener` inside `handle_event` (see PR description).

Reply:
ok I understand your point now.